src/App.tsx

```diff
@@ -1,223 +1,130 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useState,
+  useContext,
+  useEffect,
+  useRef,
+  useMemo,
+  useCallback,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessages } from './components/ErrorMessages';
+import { TodoContent } from './components/TodoContent/TodoContent';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { ErrorType } from './types/ErrorType';
+
+import { getTodos } from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+
+  const manageErrors = (errorType: ErrorType) => {
+    setErrorMessage(() => {
+      switch (errorType) {
+        case ErrorType.Endpoint:
+          return 'Fetch error';
+
+        case ErrorType.Title:
+          return 'Title can`t be empty';
+
+        case ErrorType.Add:
+          return 'Unable to add a todo';
+
+        case ErrorType.Delete:
+          return 'Unable to delete a todo';
+
+        case ErrorType.Update:
+          return 'Unable to update a todo';
+
+        case ErrorType.None:
+        default:
+          return '';
+      }
+    });
+  };
+
+  const countOfTodos = useMemo(() => todos.length, [todos]);
```

Comment:
you shouldn't cover primitive values with useMemo
we need it for objects, arrays or functions because for each re-rendering we have new link for such data type in variable
but with primitive values we don't have any link changes, we just have the simple value, so useMemo is redundant here